diff --git a/org/apache/felix/scr/impl/ComponentRegistry.java b/org/apache/felix/scr/impl/ComponentRegistry.java
index 7169f87..c9f9c0c 100644
--- a/org/apache/felix/scr/impl/ComponentRegistry.java
+++ b/org/apache/felix/scr/impl/ComponentRegistry.java
@@ -725,53 +725,6 @@ public class ComponentRegistry
 
     public void updateChangeCount()
     {
-        if ( registration != null )
-        {
-            final long count = this.changeCount.incrementAndGet();
-
-            final Timer timer;
-            synchronized ( this.changeCountTimerLock ) {
-                if ( this.changeCountTimer == null ) {
-                    this.changeCountTimer = new Timer();
-                }
-                timer = this.changeCountTimer;
-            }
-            try
-            {
-                timer.schedule(new TimerTask()
-                    {
-
-                        @Override
-                        public void run()
-                        {
-                            if ( changeCount.get() == count )
-                            {
-                                try
-                                {
-                                    registration.setProperties(getServiceRegistrationProperties());
-                                }
-                                catch ( final IllegalStateException ise)
-                                {
-                                    // we ignore this as this might happen on shutdown
-                                }
-                                synchronized ( changeCountTimerLock )
-                                {
-                                    if ( changeCount.get() == count )
-                                    {
-                                        changeCountTimer.cancel();
-                                        changeCountTimer = null;
-                                    }
-                                }
-
-                            }
-                        }
-                    }, m_configuration.serviceChangecountTimeout());
-            }
-            catch (Exception e) {
-                m_logger.log(Level.WARN,
-                    "Service changecount Timer for {0} had a problem", e,
-                    registration.getReference());
-            }
-        }
     }
 }
+/* @generated */
\ No newline at end of file
