diff --git a/org/apache/felix/cm/impl/ConfigurationManager.java b/org/apache/felix/cm/impl/ConfigurationManager.java
index 499dc8821b19..322e0dd1e214 100644
--- a/org/apache/felix/cm/impl/ConfigurationManager.java
+++ b/org/apache/felix/cm/impl/ConfigurationManager.java
@@ -1658,7 +1658,8 @@ public class ConfigurationManager implements BundleListener
 
         private void sendEvent( final int serviceIndex )
         {
-            if ( (listenerProvider[serviceIndex].getState() & (Bundle.ACTIVE | Bundle.STARTING)) > 0
+            if ( (listenerProvider[serviceIndex] != null)
+                    && (listenerProvider[serviceIndex].getState() & (Bundle.ACTIVE | Bundle.STARTING)) > 0
                     && this.listeners[serviceIndex] != null )
             {
                 Log.logger.log( LogService.LOG_DEBUG, "Sending {0} event for {1} to {2}", new Object[]
